alert-common/src/main/java/com/synopsys/integration/alert/common/action/api/AbstractJobResourceActions.java

```diff
@@ -0,0 +1,218 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.common.action.api;
+
+import java.util.Collection;
+import java.util.EnumSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
+import org.springframework.http.HttpStatus;
+
+import com.synopsys.integration.alert.common.action.ActionResponse;
+import com.synopsys.integration.alert.common.action.ValidationActionResponse;
+import com.synopsys.integration.alert.common.enumeration.ConfigContextEnum;
+import com.synopsys.integration.alert.common.enumeration.DescriptorType;
+import com.synopsys.integration.alert.common.exception.AlertDatabaseConstraintException;
+import com.synopsys.integration.alert.common.persistence.accessor.DescriptorAccessor;
+import com.synopsys.integration.alert.common.persistence.model.RegisteredDescriptorModel;
+import com.synopsys.integration.alert.common.rest.model.FieldModel;
+import com.synopsys.integration.alert.common.rest.model.JobFieldModel;
+import com.synopsys.integration.alert.common.rest.model.ValidationResponseModel;
+import com.synopsys.integration.alert.common.security.authorization.AuthorizationManager;
+
+public abstract class AbstractJobResourceActions implements JobResourceActions, ValidateAction<JobFieldModel>, TestAction<JobFieldModel> {
+    private static final EnumSet<DescriptorType> ALLOWED_JOB_DESCRIPTOR_TYPES = EnumSet.of(DescriptorType.PROVIDER, DescriptorType.CHANNEL);
+    private AuthorizationManager authorizationManager;
+    private DescriptorAccessor descriptorAccessor;
+
+    public AbstractJobResourceActions(AuthorizationManager authorizationManager, DescriptorAccessor descriptorAccessor) {
+        this.authorizationManager = authorizationManager;
+        this.descriptorAccessor = descriptorAccessor;
+    }
+
+    protected abstract Optional<JobFieldModel> findJobFieldModel(UUID id);
+
+    protected abstract ActionResponse<JobFieldModel> createAfterChecks(JobFieldModel resource);
+
+    protected abstract ActionResponse<JobFieldModel> deleteAfterChecks(UUID id);
+
+    protected abstract ActionResponse<List<JobFieldModel>> readAllAfterChecks();
+
+    protected abstract ValidationActionResponse testAfterChecks(JobFieldModel resource);
+
+    protected abstract ActionResponse<JobFieldModel> updateAfterChecks(UUID id, JobFieldModel resource);
+
+    protected abstract ValidationActionResponse validateAfterChecks(JobFieldModel resource);
+
+    private Set<String> getDescriptorNames() {
+        Set<String> descriptorNames = Set.of();
+        try {
+            descriptorNames = descriptorAccessor.getRegisteredDescriptors()
```

Comment:
@bamandel Did you create a ticket to investigate throwing `AlertDatabaseConstraintException`? I think no-args methods like this are definitely a good candidate to remove the exception.

Reply:
Yes, there's a Spike in 7.0.0 to investigate whether the exception is necessary.